src/cursor/abstract_cursor.ts

```diff
@@ -653,8 +653,15 @@ function next<T>(cursor: AbstractCursor, blocking: boolean, callback: Callback<T
 
   if (cursorId == null) {
     // All cursors must operate within a session, one must be made implicitly if not explicitly provided
-    if (cursor[kSession] == null && cursor[kTopology].hasSessionSupport()) {
-      cursor[kSession] = cursor[kTopology].startSession({ owner: cursor, explicit: false });
+    if (cursor[kSession] == null) {
+      if (cursor[kTopology].shouldCheckForSessionSupport()) {
```

Comment:
I think this is potentially not relevant, I think we always want to do the "no-op" self server selection to make sure the logicalSessionTimeoutMinutes property is correctly set by the monitor. 

Can we add some testing, maybe mock the conditions, if we start by stating the scenario it could help this code from regressing, the cursor has to be constructed without an implicit session, we could just do that part by manually constructing a cursor. Then calling next should enter serverSelection because kSession is null. And on the  recursive call of next we'll attach a session, we could just assert the session exists at the end I don't think we need to listen to the recursive call or anything that fancy.

Reply:
I've added 3 unit tests specifically on `next()`:

1. A session is set if there is a data bearing server in the cluster
2. A session is set if a data bearing server becomes available as a result of the server selection
3. No session is set if there is no data bearing server in the cluster.

As for the point about the check for session support not necessarily being relevant - I do agree that if we removed that then this would work in theory but if would produce an infinite loop on a successful server selection as `selectServer` takes a callback so we recurse on `next` when that finishes.